app/routes.py

```diff
@@ -11,131 +11,143 @@
             {
+                'id': kml_id,
+                'admin_id': kml_admin_id,
                 'success': True,
-                'id': kml_admin_id,
-                'created': item['created'],
-                'updated': item['updated'],
+                'created': timestamp,
+                'updated': timestamp,
+                'empty': empty,
                 'links':
                     {
-                        'self': f'{request.host_url}kml/{kml_admin_id}',
-                        'kml': f'{KML_STORAGE_URL}/{kml_id}'
+                        'self': f'{request.base_url}/{kml_id}',
+                        'kml': get_kml_file_link(file_key),
                     }
             }
         ),
         201
     )
 
 
-@app.route('/kml/<kml_admin_id>', methods=['GET'])
-def get_id(kml_admin_id):
-    db = get_db()  # pylint: disable=invalid-name
-    item = db.get_item(kml_admin_id)
-
+@app.route(f'/{ROUTE_ADMIN_PREFIX}/<kml_id>', methods=['GET'])
+def get_kml_metadata(kml_id):
+    item = get_db().get_item(kml_id)
     return make_response(
         jsonify(
             {
+                'id': kml_id,
                 'success': True,
-                'id': kml_admin_id,
                 'created': item['created'],
                 'updated': item['updated'],
-                'links':
-                    {
-                        'self': f'{request.host_url}kml/{item["admin_id"]}',
-                        'kml': f'{KML_STORAGE_URL}/{item["file_id"]}'
-                    }
+                'empty': item['empty'],
+                'links': {
+                    'self': request.base_url,
+                    'kml': get_kml_file_link(item['file_key']),
+                }
             }
         ),
         200
     )
 
 
-@app.route('/kml/<kml_admin_id>', methods=['PUT'])
-@validate_content_type("application/vnd.google-earth.kml+xml")
-def put_kml(kml_admin_id):
-    db = get_db()  # pylint: disable=invalid-name
-    item = db.get_item(kml_admin_id)
+@app.route(f'/{ROUTE_ADMIN_PREFIX}/<kml_id>', methods=['PUT'])
+@validate_content_type("multipart/form-data")
+@validate_content_length(KML_MAX_SIZE)
+def update_kml(kml_id):
+    db = get_db()
 
-    kml_id = item['file_id']
+    item = db.get_item(kml_id)
+    admin_id = validate_permissions(item)
 
-    quoted_str = request.get_data().decode('utf-8')
-    data = unquote_plus(quoted_str)
-    kml_string = validate_kml_string(data)
+    # Get the kml file data
+    kml_string, empty = validate_kml_file()
 
     storage = get_storage()
-    storage.upload_object_to_bucket(kml_id, kml_string)
+    storage.upload_object_to_bucket(item['file_key'], kml_string)
 
-    timestamp = datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc).isoformat()
-    db.update_item_timestamp(kml_admin_id, timestamp)
-    item = db.get_item(kml_admin_id)
+    timestamp = datetime.utcnow().replace(tzinfo=timezone.utc).isoformat()
+    db.update_item(kml_id, timestamp, empty)
```

Comment:
Does it make sense to update an existing kml, in case it is to be replaced with an empty kml string? Isn't this rather a deletion then?